### PR TITLE
point and area picking improvement for cloud names

### PR DESCRIPTION
**Pull request by larshg**
Rebased from #3763

### CODE REVIEW

File: visualization/include/pcl/visualization/area_picking_event.h [hunk 1]
```diff
@@ -62,13 +67,40 @@ namespace pcl
         {
           if (nb_points_ <= 0)
```
Comment (mvieth) [inline visualization/include/pcl/visualization/area_picking_event.h:68]: Do we even still need `nb_points_`? Here we could simply check whether `cloud_indices_` is empty or whether `indices` is empty after accumulating

File: visualization/include/pcl/visualization/point_picking_event.h [hunk 2]
```diff
@@ -55,38 +62,39 @@ namespace pcl
...
+                         CloudActorMapPtr cam_ptr,
+                         std::map<std::string, pcl::Indices>& cloud_indices) const;
+
 
       private:
-        float x_, y_, z_;
-        int idx_;
-        bool pick_first_;
+        float x_{0}, y_{0}, z_{0};
+        pcl::index_t idx_{static_cast<pcl::index_t>(-1)};
```
Comment (mvieth) [inline visualization/include/pcl/visualization/point_picking_event.h:86]: `UNAVAILABLE` from pcl_base.h would make sense here (and a few other places)

File: tools/pcd_viewer.cpp [hunk 3]
```diff
@@ -152,6 +154,18 @@ pcl::search::KdTree<pcl::PointXYZ> search;
...
 
+void
+area_callback(const pcl::visualization::AreaPickingEvent& event, void* /*cookie*/)
+{
+  const auto names = event.getCloudNames();
+
+  for (const std::string& name : names) {
+    const pcl::Indices indices = event.getPointsIndices(name);
+
+    PCL_INFO("Picked %d points from %s \n", indices.size(), name.c_str());
```
Comment (mvieth) [inline tools/pcd_viewer.cpp:165]: How about doing something useful with the indices, for example writing the points to PCD files?

Comment (larshg) [reply tools/pcd_viewer.cpp:165]: It seems to require a bit of work. Ie. I need to save the pointclouds in a map as well as handling multiples types of pointclouds (pointcloud2).
At also seems that the indices thats extracted, is not necessarily one-to-one mapping, if there is NaN points in the pointcloud.
Its a bit too much work for me - but we could add it as an issue?

**Review by mvieth — APPROVED** [review]

**Pull request merged**